src/main/java/com/bookstore/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto createOrder(OrderRequest request, User user) {
+        ShoppingCart shoppingCart = findShoppingCartByUser(user);
+
+        Order order = buildOrder(request, user, shoppingCart);
+        orderRepository.save(order);
+
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderDto> findAll(User user, Pageable pageable) {
+        Page<Order> ordersPage = orderRepository.findByUser(user, pageable);
+        List<Order> orders = ordersPage.getContent();
+        return orderMapper.toDtoList(orders);
+    }
+
+    @Transactional
+    @Override
+    public OrderDto updateOrderStatus(Long id, Order.Status status) {
+        Order order = orderRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Order not found with id: " + id));
+        order.setStatus(status);
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderItemDto> getAllOrderItems(Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Order not found with id: " + orderId));
+        return orderItemMapper.toDtoList(order.getOrderItems());
+    }
+
+    @Override
+    public OrderItemDto getOrderItem(Long orderId, Long itemId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Order not found with id: " + orderId));
+        OrderItem orderItem = order.getOrderItems().stream()
+                .filter(item -> item.getId().equals(itemId))
+                .findFirst()
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Order item not found with id: " + itemId));
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    private ShoppingCart findShoppingCartByUser(User user) {
+        return shoppingCartRepository.findByUser(user)
+                .orElseThrow(() -> new EntityNotFoundException("Shopping cart not found for user: "
+                        + user.getUsername()));
+    }
+
+    private Order buildOrder(OrderRequest request, User user, ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setShippingAddress(request.getShippingAddress());
+        order.setStatus(Order.Status.PENDING);
```

Comment:
minor comment
you can initialise PENDING status and `LocalDateTime.now()` in Order entity